### PR TITLE
Add single-issue hypertension claims to MAS integration

### PR DESCRIPTION
**Pull request by nanotone**
## Description of change
This PR updates the vets-api/MAS integration to include notifying MAS of all single-issue hypertension claims. This behavior is toggleable with a new Flipper flag. See issue link below for how it fits into the VRO project roadmap.

The bulk of this diff is actually a refactor that moves a lot of RRD-related code out of `EVSS::DisabilityCompensationForm::SubmitForm526` -- which handles all form 526 submissions to EVSS and is agnostic of RRD needs -- and into `Form526RapidReadyForDecisionConcern`. The RRD-specific behavior that gets triggered around EVSS is now exposed by the form submission as two "hooks":
- `#prepare_for_evss!`
- `#send_post_evss_notifications!`

The remainder of the RRD-specific behavior is now private to the form submission, reducing the EVSS code's [feature envy](https://github.com/troessner/reek/blob/master/docs/Feature-Envy.md).

## Original issue(s)
department-of-veterans-affairs/abd-vro#387

## Things to know about this PR
* Adds a new feature flag `rrd_hypertension_mas_notification`

## Testing
* Adds a new unit test rspec
* The team is planning to test this by first turning on the feature flag in staging/UAT

### CODE REVIEW

File: app/models/concerns/form526_rapid_ready_for_decision_concern.rb [hunk 1]
```diff
@@ -92,6 +92,30 @@ def diagnostic_codes
...
+  def prepare_for_evss!
+    return unless forward_to_mas?
+
+    save_metadata(forward_to_mas: true)
+    insert_classification_codes
+  end
+
+  def send_post_evss_notifications!
+    send_rrd_completed_notification if rrd_job_selector.rrd_applicable?
+    notify_mas if form.dig('rrd_metadata', 'forward_to_mas')
```
Comment (yoomlam) [inline app/models/concerns/form526_rapid_ready_for_decision_concern.rb:104]: Wondering if we should have a `read_metadata('forward_to_mas')` (which calls `form.dig('rrd_metadata', 'forward_to_mas')`)  to complement `save_metadata(forward_to_mas: true)`.

**Review by yoomlam — APPROVED** [review]
LGTM. It may be helpful for reading the code to make some of the methods in the Concern `private` to make it more obvious they're only used within the concern.

File: app/models/concerns/form526_rapid_ready_for_decision_concern.rb (hunk 1 above)
Comment (nanotone) [reply app/models/concerns/form526_rapid_ready_for_decision_concern.rb:105]: Yep, that's a great idea. I'll try to do it in another PR that doesn't already contain a boatload of refactoring.

**Review by sethdarragile6 — APPROVED** [review]
lgtm

**Review by holdenhinkle — APPROVED** [review]

**Pull request merged**